File: ruby-elasticemail-examples/examples/statistics.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

##
# Statistics
#
# Account-wide sending statistics for the last 30 days.
#
# Usage: ruby examples/statistics.rb

require_relative "ee"

statistics_api = ElasticEmail::StatisticsApi.new

# The SDK documents `from`/`to` as Time, but it passes the values straight into the
# query string. Time#to_s is not the format the API expects (YYYY-MM-DDThh:mm:ss, UTC),
# so format the strings here.
iso = ->(t) { t.utc.strftime("%Y-%m-%dT%H:%M:%S") }
to_time = Time.now
from_time = to_time - 30 * 24 * 60 * 60

begin
  data = statistics_api.statistics_get(iso.call(from_time), to: iso.call(to_time))

  puts "=== Statistics #{iso.call(from_time)} to #{iso.call(to_time)} ==="
  puts "Recipients:    #{data.recipients}"
  puts "Emails total:  #{data.email_total}"
  puts "Delivered:     #{data.delivered}"
  puts "Bounced:       #{data.bounced}"
  puts "In progress:   #{data.in_progress}"
  puts "Opened:        #{data.opened}"
  puts "Clicked:       #{data.clicked}"
  puts "Unsubscribed:  #{data.unsubscribed}"
  puts "Complaints:    #{data.complaints}"
rescue ElasticEmail::ApiError => e
  api_error("fetch statistics", e)
end

File: python-elasticemail-examples/examples/statistics.py
"""
Statistics Example

Account-wide sending statistics for the last 30 days.
The SDK takes datetime objects and serializes them as YYYY-MM-DDThh:mm:ss (UTC).

Usage: python examples/statistics.py
"""

import datetime
import os
import sys

import ElasticEmail

sys.path.insert(0, os.path.dirname(__file__))
from ee import get_configuration, print_api_error


def main():
    configuration = get_configuration()
    to_date = datetime.datetime.utcnow().replace(microsecond=0)
    from_date = to_date - datetime.timedelta(days=30)

    with ElasticEmail.ApiClient(configuration) as api_client:
        statistics_api = ElasticEmail.StatisticsApi(api_client)

        try:
            data = statistics_api.statistics_get(from_date, to=to_date)

            print("=== Statistics {} to {} ===".format(from_date.isoformat(), to_date.isoformat()))
            print("Recipients:   ", data.recipients)
            print("Emails total: ", data.email_total)
            print("Delivered:    ", data.delivered)
            print("Bounced:      ", data.bounced)
            print("In progress:  ", data.in_progress)
            print("Opened:       ", data.opened)
            print("Clicked:      ", data.clicked)
            print("Unsubscribed: ", data.unsubscribed)
            print("Complaints:   ", data.complaints)
        except ElasticEmail.ApiException as e:
            print_api_error("fetch statistics", e)
            sys.exit(1)


if __name__ == "__main__":
    main()
